core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -457,6 +468,12 @@ public Frame decorrelateRel(Aggregate rel) {
     // Aggregate itself should not reference corVars.
     assert !cm.mapRefRelToCorRef.containsKey(rel);
 
+    for (AggregateCall call : rel.getAggCallList()) {
+      if (aggFunctionNotFitToDecorrelate(call.getAggregation())) {
+        return null;
+      }
+    }
```

Comment:
Should we allow count agg for subquery where the group key is not empty? e.g.
`select * from foo where (a, b) in (select c, count(*) from bar where bar.d=foo.d group by c)`

Reply:
But I am not sure Calcite can decorrelate it or not.

Reply:
Rather than failing the decorrelation when COUNT, do we consider just disable decorrelation when subquery is an Aggregate?
Because user have the flexibility to define UDAF, it's hard to predicate the Agg behavior or enumerate all types of Agg functions.

Reply:
@yanlin-Lynn 
If the PR is already for review, could you please remove `[WIP]` from PR title ?

Reply:
@hsyuan , yes, we can allow that when group key is not empty. I add a test case for that.

@jinxing64 RelDecorrelator.decorrelateQuery is a public method, user may call it directly. So, I would perfer to fail the decorrelation. But, your concern is valid, so I add the check for UDAF. How about that.

Reply:
Thanks @yanlin-Lynn 
I would hear committers' advice.